server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java

```diff
@@ -474,6 +479,7 @@ public KafkaEventQueue(
             this.eventHandler, false);
         this.shuttingDown = false;
         this.interrupted = false;
+        this.idleTimeCallback = idleTimeCallback != null ? idleTimeCallback : __ -> { };
```

Comment:
I suggest not using `null` in public interfaces. Please use `Optional<T>` if a parameter is optional.
FYI: https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/